controllers/topology/internal/contract/controlplane.go

```diff
@@ -63,6 +70,81 @@ func (c *ControlPlaneContract) Replicas() *ControlPlaneReplicas {
 	return &ControlPlaneReplicas{}
 }
 
+// StatusReplicas provide access to status.replicas field  in a ControlPlane object, if any.
+func (c *ControlPlaneContract) StatusReplicas() *ControlPlaneStatusReplicas {
+	return &ControlPlaneStatusReplicas{}
+}
+
+// UpdatedReplicas provide access to status.updatedReplicas field  in a ControlPlane object, if any.
+func (c *ControlPlaneContract) UpdatedReplicas() *ControlPlaneUpdatedReplicas {
+	return &ControlPlaneUpdatedReplicas{}
+}
+
+// ReadyReplicas provide access to status.readyReplicas field  in a ControlPlane object, if any.
+func (c *ControlPlaneContract) ReadyReplicas() *ControlPlaneReadyReplicas {
+	return &ControlPlaneReadyReplicas{}
+}
+
+// IsUpgrading returns true if the control plane is in the middle of an upgrade, false otherwise.
+// A control plane is considered upgrading if:
+// - if status.version is not yet set
+// - if spec.version and status.verison do not match.
+func (c *ControlPlaneContract) IsUpgrading(obj *unstructured.Unstructured) (bool, error) {
+	specVersion, err := c.Version().Get(obj)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get control plane spec version")
+	}
+	statusVersion, err := c.StatusVersion().Get(obj)
+	if err != nil {
+		if errors.Is(err, errNotFound) { // status version is not yet set
+			return true, nil
+		}
```

Comment:
Should we return error or false in this case?
If there is no status version, we can't really determine if we are upgrading.... also, IsScaling should provide us a better signal for preventing upgrades during the initial provisioning phase